### ISSUE TITLE
MNTP: Exception with orphan node IDs

### ISSUE THREAD

**Issue by tomfulton** [Jeavon/Umbraco-Core-Property-Value-Converters#1]
One of my MNTP's has references to a node that no longer exists, which seems to cause the below exception when attempting to access:

![ex](https://cloud.githubusercontent.com/assets/1396376/4728512/ef23dcc8-5976-11e4-9a2e-21f26dbd2c35.png)

I think the exception is coming from the EntityService since we're passing it an invalid node ID.

I started to look at fixing this but wasn't sure how else we'd determine the `UmbracoObjectType` - other than a loop with some try/catches :grimacing: 

Any ideas, have you ran into this before?

**Comment by Jeavon** [comment]
GetObjectType should return null for a deleted nodeId, if it's throwing then it's a core bug :) I'll have a look at core but probably best to add a try catch if it's throwing currently.

**Comment by Jeavon** [comment]
@tomfulton I've tried to replicate with a unpublished and deleted node without success, any tips on how to replicate?

**Comment by tomfulton** [comment]
Hey @jeavon, I can replicate by passing a non-existing ID to:

``` csharp
ApplicationContext.Current.Services.EntityService.GetObjectType(3333333);
```

I was also able to reproduce in the Umbraco UI (7.1.6).  It should happen once you delete the node from the Recycle Bin, and it's still selected on an MNTP.  Is it not happening for you?

I think you're right that the core shouldn't throw for this, but even if they don't we may still have the same issue.  Right now the PVC relies on this method to figure out what type of nodes it should request/return (Document/Media/Member).  If the first node ID in the list is invalid (this was my case), the PVC won't know how to process the rest.

Maybe if the first ID doesn't exist, we fallback to the next, and so-on?  And if all are invalid we return an empty list?

**Comment by Jeavon** [comment]
Fixed (worked around) in v2.1.3

**Comment by tomfulton** [comment]
@Jeavon Thanks, works like a champ!  :beers:

Hope to see this fixed in the core too.  I'm following your [pull request](https://github.com/umbraco/Umbraco-CMS/pull/520) and [U4-5698](http://issues.umbraco.org/issue/U4-5698), nice work!

**Comment by Jeavon** [comment]
Me too, think my workaround isn't very efficient but now I have a good fix for core, so hopefully Umbraco v7.2.1 I can revert :)
